Backpack-SwiftUI/InsetBanner/Classes/BPKSponsoredInsetBanner.swift

```diff
@@ -0,0 +1,195 @@
+        self.callToAction = callToAction
+        self.variant = variant
+        self.backgroundColor = backgroundColor
+        self.customAccessibilityLabel = customAccessibilityLabel
+        self.image = image
+    }
+    
+    public init(
+        logoContent: LogoContent,
+        callToAction: CallToAction,
+        variant: Variant,
+        backgroundColor: Color,
+        customAccessibilityLabel: String? = nil,
+        image: Image? = nil
+    ) {
+        self.logo = logoContent
+        self.title = nil
+        self.subheadline = nil
+        self.callToAction = callToAction
+        self.variant = variant
+        self.backgroundColor = backgroundColor
+        self.customAccessibilityLabel = customAccessibilityLabel
+        self.image = image
+    }
+    
+    public var body: some View {
+        VStack(spacing: 0) {
+            topView
+                .background(backgroundColor)
+                .zIndex(1)
+        }
+        .background(
+            RoundedRectangle(cornerRadius: .sm, style: .continuous)
+                .foregroundColor(.sponsoredBannerBackgroundColor)
+        )
+        .clipShape(
+            RoundedRectangle(cornerRadius: .sm, style: .continuous)
+        )
+        .if(customAccessibilityLabel != nil) { view in
+            view
+                .accessibilityElement(children: .ignore)
+                .accessibilityLabel(customAccessibilityLabel ?? "")
+        }
+    }
+    
+    private var topView: some View {
+        VStack() {
+            HStack(spacing: .md) {
+                titlesView
+                Spacer()
+                VStack {
+                    Button(action: callToAction.onClick) {
+                        buttonContent
+                    }
+                    Spacer()
+                }
+            }
+            .padding(.base)
+            .accessibilityElement(children: .combine)
+            .buttonStyle(SponsoredInsetBannerButtonStyle(
+                foregroundColor: (variant == .onDark) ?
+                    Color(BPKColor.textOnDarkColor) :
+                    Color(BPKColor.textOnLightColor),
+                backgroundColor: backgroundColor
+            ))
+            if let image = image {
+                image
+                    .resizable()
+                    .scaledToFill()
+                    .frame(height: 120)
+                    .clipped()
+            }
+        }
+    }
+    
+    private var buttonContent: some View {
+        HStack {
+            BPKText(callToAction.text, style: .caption)
+                .foregroundColor(variant.color)
+                .lineLimit(nil)
+            iconView
+        }
+    }
+    
+    private var titlesView: some View {
+        VStack(alignment: .leading, spacing: .sm) {
+            logo
+                .frame(maxWidth: Constants.maxLogoWidth, maxHeight: Constants.maxLogoHeight)
+                .fixedSize(horizontal: true, vertical: false)
+            if let title = title {
```

Comment:
Small nit: we can just do: 

```suggestion
            if let title {
```